src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -561,106 +541,79 @@ get_relation_constraint_oid_compat(HeapTuple heapTuple)
 
 
 /*
- * HasForeignKeyToReferenceTable function scans the pgConstraint table to
- * fetch all of the constraints on the given relationId and see if at least one
- * of them is a foreign key referencing to a reference table.
+ * HasForeignKeyToReferenceTable function returns true if any of the foreign
+ * key constraints on the relation with relationOid references to a reference
+ * table.
  */
 bool
-HasForeignKeyToReferenceTable(Oid relationId)
+HasForeignKeyToReferenceTable(Oid relationOid)
 {
-	ScanKeyData scanKey[1];
-	int scanKeyCount = 1;
-	bool hasForeignKeyToReferenceTable = false;
+	bool findAll = false;
+	return ForeignKeysToReferenceTables(relationOid, findAll) != NIL;
+}
 
-	Relation pgConstraint = heap_open(ConstraintRelationId, AccessShareLock);
-	ScanKeyInit(&scanKey[0], Anum_pg_constraint_conrelid, BTEqualStrategyNumber, F_OIDEQ,
-				relationId);
-	SysScanDesc scanDescriptor = systable_beginscan(pgConstraint,
-													ConstraintRelidTypidNameIndexId,
-													true, NULL,
-													scanKeyCount, scanKey);
 
-	HeapTuple heapTuple = systable_getnext(scanDescriptor);
-	while (HeapTupleIsValid(heapTuple))
+/*
+ * ForeignKeysToReferenceTables function returns list of OIDs for the foreign
+ * key constraints on the given relationId that are referencing to reference
+ * tables.
+ */
+static List *
+ForeignKeysToReferenceTables(Oid relationOid, bool findAll)
```

Comment:
I think it is more clear to call this function `ForeignKeyOidsToReferenceTables` so that we know that this function returns a list of oids.